src/services/SnapshotService.ts

```diff
@@ -3,20 +3,79 @@ import isNumber from 'lodash/isNumber'
 
 import { SnapshotApi, SnapshotReceipt } from '../clients/SnapshotApi'
 import { SnapshotGraphql } from '../clients/SnapshotGraphql'
-import { DetailedScores, SnapshotProposal, SnapshotVote, VpDistribution } from '../clients/SnapshotGraphqlTypes'
+import {
+  DetailedScores,
+  ServiceHealth,
+  ServiceStatus,
+  SnapshotProposal,
+  SnapshotStatus,
+  SnapshotVote,
+  VpDistribution,
+} from '../clients/SnapshotTypes'
 import * as templates from '../entities/Proposal/templates'
 import { proposalUrl, snapshotProposalUrl } from '../entities/Proposal/utils'
 import { SNAPSHOT_SPACE } from '../entities/Snapshot/constants'
 import { isSameAddress } from '../entities/Snapshot/utils'
 import { inBackground } from '../helpers'
 import { Avatar } from '../utils/Catalyst/types'
+import { ErrorCategory } from '../utils/errorCategories'
 
+import CacheService from './CacheService'
+import { ErrorService } from './ErrorService'
 import { ProposalInCreation, ProposalLifespan } from './ProposalService'
 import RpcService from './RpcService'
 
 const DELEGATION_STRATEGY_NAME = 'delegation'
+const SLOW_RESPONSE_TIME_THRESHOLD_IN_MS = 8000 // 8 seconds
+const STATUS_CACHE_TIMEOUT_IN_SECONDS = 60 // 1 minute
 
 export class SnapshotService {
+  public static async getStatus(): Promise<SnapshotStatus | undefined> {
+    try {
+      const cachedStatus = CacheService.get<SnapshotStatus>('snapshotStatus')
+      if (cachedStatus) {
+        return cachedStatus
+      }
+
+      const snapshotStatus = await this.sense()
+      CacheService.set('snapshotStatus', snapshotStatus, STATUS_CACHE_TIMEOUT_IN_SECONDS)
+
+      logger.log('Snapshot status:', snapshotStatus)
+      return snapshotStatus
+    } catch (error) {
+      ErrorService.report('Unable to determine snapshot status', { error, category: ErrorCategory.Snapshot })
+      return undefined
+    }
+  }
+
+  private static async sense(): Promise<SnapshotStatus> {
```

Comment:
I think we could go with the "ping" nomenclature as this is used internally in this method by the snapshot clients and we could infer a ping is a status check.